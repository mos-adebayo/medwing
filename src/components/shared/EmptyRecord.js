import React from 'react';

const EmptyRecord = ({message}) => {
    return (
        <div className={'section empty-container'}>
            <h1 className={'center-align'}>
                <i className="large material-icons">info</i>
            </h1>
            <p className={'center-align'}>
                {message}
            </p>
        </div>
    );
};

export default EmptyRecord;
