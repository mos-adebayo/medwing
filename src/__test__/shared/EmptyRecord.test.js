import React from "react";
import { configure, shallow } from "enzyme";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import EmptyRecord from "../../components/shared/EmptyRecord";


//Enzyme Configurations
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

const message = 'No data yet!';

test('EmptyRecord components snapshot', () => {
    const component = renderer.create(
        <EmptyRecord message={message} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('EmptyRecord component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<EmptyRecord message={message} />, div);
    ReactDOM.unmountComponentAtNode(div);
});


test("EmptyRecord component receives props and then renders text", () => {
    const ComponentErrorMessageRender = shallow(<EmptyRecord message={message} />);
    expect(ComponentErrorMessageRender.contains((
        <h1 className={'center-align'}>
            <i className="large material-icons">info</i>
        </h1>
    ))).toBeTruthy();
    expect(ComponentErrorMessageRender.contains((<p className={'center-align'}>{message}</p>))).toBeTruthy()
});
