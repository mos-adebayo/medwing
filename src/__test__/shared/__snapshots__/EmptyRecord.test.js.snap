// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyRecord components snapshot 1`] = `
<div
  className="section empty-container"
>
  <h1
    className="center-align"
  >
    <i
      className="large material-icons"
    >
      info
    </i>
  </h1>
  <p
    className="center-align"
  >
    No data yet!
  </p>
</div>
`;
